### PR TITLE
fix: send Slack notification on last retry failure

### PR DESCRIPTION
**Pull request by metakuni**
### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

File: backend/corpora/dataset_processing/process.py [hunk 1]
```diff
@@ -576,19 +573,9 @@ def main():
         )
         return_value = 1
 
-    if return_value > 0:
-        notify_slack_failure(dataset_id)
     return return_value
 
 
-def notify_slack_failure(dataset_id):
```
Comment (metakuni) [inline backend/corpora/dataset_processing/process.py:584]: Moved to `backend/corpora/dataset_processing/slack.py`

File: backend/corpora/dataset_processing/process.py [hunk 2]
```diff
@@ -576,19 +573,9 @@ def main():
         )
         return_value = 1
 
-    if return_value > 0:
-        notify_slack_failure(dataset_id)
```
Comment (metakuni) [inline backend/corpora/dataset_processing/process.py:580]: Moved to `backend/corpora/upload_failures/app.py`

File: backend/corpora/dataset_processing/slack.py [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+from backend.corpora.common.corpora_config import CorporaConfig
 from backend.corpora.common.entities import Dataset
 from backend.corpora.common.utils.db_session import db_session_manager
 from backend.corpora.common.utils.json import CustomJSONEncoder
 
+logger = logging.getLogger(__name__)
+
 
 def format_slack_message(dataset_id):
+    # TODO: pass this info in so that the Slack message formatter doesn't have to know about the db
```
Comment (metakuni) [inline backend/corpora/dataset_processing/slack.py:17]: I'll address this when I work on #1586

File: backend/corpora/dataset_processing/slack.py [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
+
+from backend.corpora.common.corpora_config import CorporaConfig
 from backend.corpora.common.entities import Dataset
 from backend.corpora.common.utils.db_session import db_session_manager
 from backend.corpora.common.utils.json import CustomJSONEncoder
 
+logger = logging.getLogger(__name__)
+
 
 def format_slack_message(dataset_id):
```
Comment (metakuni) [inline backend/corpora/dataset_processing/slack.py:16]: Refactor: Moved from `backend/corpora/dataset_processing/process.py`

File: backend/corpora/dataset_processing/slack.py [hunk 5]
```diff
@@ -48,3 +55,14 @@ def format_slack_message(dataset_id):
         ]
     }
     return json.dumps(data, indent=2)
+
+
+def notify_slack_failure(dataset_id: str) -> None:
```
Comment (metakuni) [inline backend/corpora/dataset_processing/slack.py:60]: Refactor: Moved from `backend/corpora/dataset_processing/process.py`

File: backend/corpora/upload_failures/app.py [hunk 6]
```diff
@@ -3,13 +3,20 @@
...
+
+    deployment_stage = os.environ["DEPLOYMENT_STAGE"]
+
+    cellxgene_bucket = f"hosted-cellxgene-{deployment_stage}"
     if os.getenv("CELLXGENE_BUCKET"):
         cellxgene_bucket = os.getenv("CELLXGENE_BUCKET")
     delete_many_from_s3(cellxgene_bucket, object_key)
     update_dataset_processing_status_to_failed(dataset_uuid, event["error"]["Cause"])
+
+    notify_slack_failure(dataset_uuid)
```
Comment (metakuni) [inline backend/corpora/upload_failures/app.py:22]: Refactor: Moved from `backend/corpora/dataset_processing/process.py`

**Comment by metakuni** [comment]
Leaving this draft PR open. Work to be resumed during next cool down.

**Comment by maniarathi** [comment]
> Leaving this draft PR open. Work to be resumed during next cool down.

Any particular reason to leave the PR open? I'm thinking that it would be good to make sure that we don't leave open long-lived PRs and instead opt for long lived branches that you could keep rebasing with `main`.

**Pull request closed without merging**